src/Microsoft.OData.Core/JsonLight/ODataJsonLightPropertyAndValueDeserializer.cs

```diff
@@ -728,6 +703,67 @@ in resourceState.PropertyAndAnnotationCollector.GetODataPropertyAnnotations(nest
+        /// Reads a nested stream
+        /// </summary>
+        /// <param name="resourceState">The state of the reader for resource to read.</param>
+        /// <param name="property">The property for which to read the nested resource info. null for undeclared property.</param>
+        /// <param name="propertyName">The property name.</param>
+        /// <returns>The nested resource info for the stream collection.</returns>
+        /// <remarks>
+        /// This method doesn't move the reader.
+        /// </remarks>
+        protected static ODataJsonLightReaderNestedInfo ReadStreamNestedResourceInfo(IODataJsonLightReaderResourceState resourceState, IEdmStructuralProperty property, string propertyName)
+        {
+            Debug.Assert(resourceState != null, "resourceState != null");
+            Debug.Assert((propertyName != null) || (property != null),
+                "The collection property and property name shouldn't both be null");
+            Debug.Assert(property == null || property.Type.Definition.IsStream(),
+                "The type of the property, if specified, must be Edm.Stream");
+
+            ODataNestedResourceInfo nestedResourceInfo = new ODataNestedResourceInfo()
+            {
+                Name = propertyName,
+                IsCollection = false,
+                IsComplex = false
+            };
+
+            return ODataJsonLightReaderNestedResourceInfo.CreateResourceReaderNestedResourceInfo(nestedResourceInfo, property, EdmCoreModel.Instance.GetPrimitiveType(EdmPrimitiveTypeKind.Stream));
+        }
+*/
+
+        /// <summary>
+        /// Reads a nested stream collection as a resource set.
+        /// </summary>
+        /// <param name="resourceState">The state of the reader for resource to read.</param>
+        /// <param name="collectionProperty">The collection of stream property for which to read the nested resource info. null for undeclared property.</param>
+        /// <param name="propertyName">The property name.</param>
+        /// <param name="elementType">They primitive type of the collection element</param>
+        /// <returns>The nested resource info for the stream collection.</returns>
+        /// <remarks>
+        /// This method doesn't move the reader.
+        /// </remarks>
+        protected static ODataJsonLightReaderNestedResourceInfo ReadStreamCollectionNestedResourceInfo(IODataJsonLightReaderResourceState resourceState, IEdmStructuralProperty collectionProperty, string propertyName, IEdmType elementType)
+        {
+            Debug.Assert(resourceState != null, "resourceState != null");
+            Debug.Assert((propertyName != null) || (collectionProperty != null),
+                "The collection property and property name shouldn't both be null");
+
+            ODataNestedResourceInfo nestedResourceInfo = new ODataNestedResourceInfo()
+            {
+                Name = propertyName,
+                IsCollection = true,
+                IsComplex = false

```

Comment:
>IsComplex = false [](start = 16, length = 17)

is "IsComplex = false" always true?

Reply:
Yes; because it's a stream, not a complex.  It's not an entity either, but we don't have a property for that.

---
In reply to: [259082576](https://github.com/OData/odata.net/pull/1204#discussion_r259082576) [](ancestors = 259082576)